### PR TITLE
Tray icon

### PR DESCRIPTION
**Pull request by rafaelcastrocouto**
Added a tray icon with close option for the server.
In order to avoid multiple Nodejs tasks I wrapped Yez inside Electron since it's required for the tray icon.

Soon I will add menu options to change icon style and to control Yez! tasks directly from the tray icon menu.

### CODE REVIEW

**Comment by krasimir** [comment]
Can you please explain what exactly "wrapped Yez inside Electron" means. 

The idea of Yez is to work in Chrome as an extension. Isn't it wrapping it as an electron app means that it will be used outside of the browser. If yes then I'll suggest that we keep the extension and create a folder clone.

File: chrome/manifest.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "manifest_version": 2,
     "name": "Yez",
     "description": "Task runner",
-    "version": "0.1.10",
+    "version": "0.1.9",
```
Comment (krasimir) [inline chrome/manifest.json]: Sorry, why is that change? Decreasing the minor version.

**Comment by rafaelcastrocouto** [comment]
The server part of Yez! was wrapped with Electron ... 
It's easy to understand by looking at "node/index.js" 
Please forgive me about the version problem ... I did not update my fork before committing.
You can close this pull, I'm gonna fix everything and make a new one.

**Comment by krasimir** [comment]
> You can close this pull, I'm gonna fix everything and make a new one

Thanks. It is also a little bit unclear what will be the final result. Is your idea to run the backend of Yez! via electron or that's just an addition.

**Comment by rafaelcastrocouto** [comment]
Yes, the idea is to run the backend via electron.
First, I implemented the tray icon separated from the backend, but I realized it was a waste of memory to have both, node and electron, running at the same time. So I wrapped the backend in electron with the tray icon.
Besides the tray icon, electron will also make easy to generate an installer for windows or any other OS.
I still plan to make a standalone electron window for Yez!, but it seems very tricky because of the background work that happens in the chrome extension.
I tried to use [devtool](https://github.com/Jam3/devtool) to avoid this problem, but [yez does not work inside it](https://github.com/electron/electron/issues/4698). Seems it's a [react-devtools](https://github.com/electron/electron/issues/915) problem.
I'll try to use [nw.js](http://nwjs.io/) to fix this since they have react-devtools support.
I'll also make a theme switcher for the light and dark themes.

**Comment by krasimir** [comment]
Everything sounds really good to me. I would like to have both in parallel though. I mean still provide a standalone node.js module because there are people which are using it and prefer running Yez! from the terminal.

**Comment by rafaelcastrocouto** [comment]
Hey @krasimir , 

I implemented the tray icon in parallel and the dark theme as options.
`yez --dark --tray` will start Yez! with dark theme and tray icon.
Both can be changed at the extension home screen.
If you feel like, you can check out [my fork](https://github.com/rafaelcastrocouto/yez).
I've got a few questions about what I'm gonna do next.

1 - Would you like me to put all style in a css file and make absurdjs work with classes only? 
(I'm not a fan of mixing css and js)

2 - Do you have any reason to work with event pages (background.js) instead of using the back-end (sockets)? I would like to move all synchronization to the back-end, so an electron standalone window would work. With this, I could make an static server for the chrome folder, to enable users to have Yez! in a common tab an even make a favorite button.

So what do you think?

**Comment by krasimir** [comment]
I agree with both points `1` and `2`. I'll suggest to keep `yez` running the node.js version of the backend. The one that works now and make `yez --e` to run electron's app.

**Comment by rafaelcastrocouto** [comment]
Ok ...  both implemented ... electron's app will only run if the user turn the tray icon on.
Now there's a savedTasks.json to replace the local storage.
Next I would like to change Yez style in order to match devtool's style, what's your thoughts on this?

**Comment by krasimir** [comment]
> Next I would like to change Yez style in order to match devtool's style, what's your thoughts on this?
Oh ... this will take time. I guess it is a matter of CSS styling.

**Comment by rafaelcastrocouto** [comment]
It's done ... could you please test it out before I pull this ... I gotta feeling you might wanna make a few suggestions.
What to test:
 - new theme (can be changed from home screen options or with `yez --dark`)
 - tray icon (can be enabled from home screen options or with `yez --tray`)
 - tasks management (the back-end is handling this and I might have missed something)
 - aliases saving (it's also on the back-end)

I also made a `dir` command to list directories. I did this cuz node don't seem to receive any output from window's `dir`. It got me wandering, does `ls` works fine in other OS?

**Comment by krasimir** [comment]
Thanks @rafaelcastrocouto. I'll test it these days and will let you know.

**Comment by krasimir** [comment]
Hi @rafaelcastrocouto,

I'm just testing this PR. Here are few things:

* `http-server` is missing in `package.json` and the backend fails. What is the idea behind it?
* In `tray.js` you are assuming that the electron app is running always with arguments and this fails `argv = JSON.parse(process.argv[2]),`
* There is no documentation of how to run the electron app. NPM script `npm run electron` will be nice.

Otherwise it looks nice.

**Comment by rafaelcastrocouto** [comment]
Thx a lot for your time!

`http-server` is meant to make the `chrome` folder accessible at `localhost:port`.It can be replaced for inline code, another module or removed at all. What's your call?

There's no need to call `tray.js` directly, if the user doesn't start with `--tray`, it can be enabled through the home screen option. There's a little problem in making it possible to run `tray.js` directly ... when the user click the close button, Yez won't really close, just the electron app, so the user must have the understatement that there are parallel processes (not a good thing IMHO).

**Comment by krasimir** [comment]
> http-server is meant to make the chrome folder accessible at localhost:port.It can be replaced for inline code, another module or removed at all. What's your call?

Sorry, but it's not clear to me why the `chrome` folder should be accessible through http.

> about tray.js

To be honest I never used electron and what I try was just running ./node_modules/.bin/`electron ./electron` and because `tray.js` is set as to `main` property I got the error.

**Comment by rafaelcastrocouto** [comment]
In the tray menu, there's an option to open "Yez! in browser", that's why I'm serving `chrome` with the `http-server` module (this way the user can bookmark Yez).

The main property in indeed suppose to be used that way, but as I said before, if the user calls `tray.js`, there will be no link between the "Yez node back end" and the "electron tray app". That's why the "electron app" should always be called by the back-end, either with the `--tray` argument or through the home screen option.

So, what should I change before making a new pull?

**Pull request closed without merging**